src/Imdb/Title.php

```diff
@@ -477,29 +477,46 @@ public function comment_split() {
  #-------------------------------------------------------[ Recommendations ]---
   /**
    * Get recommended movies (People who liked this...also liked)
-   * @return array recommendations (array[imdbid,title,year])
+   * @return array recommendations (array[title,imdbid,year,endyear,type])
    * @see IMDB page / (TitlePage)
    */
-  public function movie_recommendations() {
-    if (empty($this->movierecommendations)) {
-      $doc = new \DOMDocument();
-      @$doc->loadHTML($this->getPage("Title"));
-      $xp = new \DOMXPath($doc);
-      $cells = $xp->query("//div[@id=\"title_recs\"]/div[@class=\"rec_overviews\"]/div[@class=\"rec_overview\"]/div[@class=\"rec_details\"]");
-      foreach ($cells as $cell) {
-        preg_match('!tt(\d+)!',$cell->getElementsByTagName('a')->item(0)->getAttribute('href'),$ref);
-        $movie['title'] = trim($cell->getElementsByTagName('a')->item(0)->nodeValue);
-        $movie['imdbid'] = $ref[1];
-        preg_match('!(\d+)!',$cell->getElementsByTagName('span')->item(0)->nodeValue,$ref);
-        if (!isset($ref[1])) {
-          preg_match('!(\d+)!',$cell->getElementsByTagName('span')->item(1)->nodeValue,$ref);
-        }
-        $movie['year'] = $ref[1];
-        $this->movierecommendations[] = $movie;
-      }
-    }
-    return $this->movierecommendations;
-  }
+public function movie_recommendations() {
+	if (empty($this->movierecommendations)) {
+		$doc = new \DOMDocument();
+		@$doc->loadHTML($this->getPage("Title"));
+		$xp = new \DOMXPath($doc);
+		if ($cells = $xp->query("//div[@id=\"title_recs\"]/div[@class=\"rec_overviews\"]/div[@class=\"rec_overview\"]/div[@class=\"rec_details\"]/div[@class=\"rec-info\"]/div[@class=\"rec-jaw-upper\"]/div[@class=\"rec-title\"]")){
```

Comment:
This is sooo complicated. Would ```"//div[@id=\"title_recs\"]//div[@class=\"rec-title\"]"``` not cover it?

Reply:
Also, the ```if``` does nothing but add more indentation. $cells is always an iterable.

Reply:
The only thing i know about domxpath query is that it only works with a fill directory path so that is why i made it this way.

So it should be possible to shorten this, so i will try that. Thanks

Reply:
reworked the xpath query and removed the unnecessary if